src/dodal/devices/scintillator.py

```diff
@@ -1,10 +1,81 @@
-from ophyd_async.core import StandardReadable
+from math import isclose
+
+from ophyd_async.core import Reference, StandardReadable, StrictEnum, derived_signal_rw
 from ophyd_async.epics.motor import Motor
 
+from dodal.common.beamlines.beamline_parameters import GDABeamlineParameters
+from dodal.devices.aperturescatterguard import ApertureScatterguard, ApertureValue
+
+
+class InOut(StrictEnum):
+    """Currently Hyperion only needs to move the scintillator out for data collection."""
+
+    OUT = "Out"
+    UNKNOWN = "Unknown"
+
 
 class Scintillator(StandardReadable):
-    def __init__(self, prefix: str, name: str = ""):
+    """Moves a scintillator into and out of the beam.
+
+    The scintillator will light up when hit with xrays, this allows scientists to use it
+    in conjunction with the optical OAV camera to commission the beamline.
+
+    When moved out of the beam it is parked under the table. This parking has a potential
+    to collide with the aperture/scatterguard if that is not correctly parked already.
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        aperture_scatterguard: Reference[ApertureScatterguard],
+        beamline_parameters: GDABeamlineParameters,
+        name: str = "",
+    ):
         with self.add_children_as_readables():
-            self.y = Motor(prefix + "-MO-SCIN-01:Y")
-            self.z = Motor(prefix + "-MO-SCIN-01:Z")
+            self.y_mm = Motor(prefix + "-MO-SCIN-01:Y")
+            self.z_mm = Motor(prefix + "-MO-SCIN-01:Z")
```

Comment:
As above, move these onto the prefix of the device instance rather than on the class